fs/copy_linux.go

```diff
@@ -49,22 +50,33 @@ func copyFileContent(dst, src *os.File) error {
 		return errors.Wrap(err, "unable to stat source")
 	}
 
-	n, err := unix.CopyFileRange(int(src.Fd()), nil, int(dst.Fd()), nil, int(st.Size()), 0)
-	if err != nil {
-		if err != unix.ENOSYS && err != unix.EXDEV {
-			return errors.Wrap(err, "copy file range failed")
+	var written int64
+	size := st.Size()
+	first := true
+
+	for written < size {
+		var desired int
+		if size-written > math.MaxInt32 {
```

Comment:
OK in fact I think we should not do something like this instead:

```golang
sfd := int(src.Fd())
dfd := int(dst.Fd())
size := st.Size()
for size > 0 {
		n, err := unix.CopyFileRange(sfd, nil, dfd, nil, int(size), 0)
		if err != nil {
				// ...
		}
		size -= n
}
```

This way we don't have any knowledge of how big the size can be, but just relying on the fact that the syscall returns the number of bytes actually copied.

Note that in here `int(size)` can indeed be negative, but it is internally used as `size_t`, ie an unsigned type.

Reply:
Yeah, I was wondering why it didn't fail before when I did this. This makes sense.

Reply:
Done.